src/github.com/matrix-org/dendrite/roomserver/types/types.go

```diff
@@ -75,3 +83,15 @@ const (
 	// EmptyStateKeyNID is the numeric ID for the empty state key.
 	EmptyStateKeyNID = 1
 )
+
+// StateDataNIDList is used to return the result of bulk StateDataNID lookups from the database.
+type StateDataNIDList struct {
```

Comment:
The comment does indeed explain it, but that's not the point. The point is to try to make it obvious when it is used (and you only have the type to go by) that this is just a convenience struct rather than something which we may want to pass through to other APIs.

I'm open to other naming suggestions.